src/sympc/tensor/mpc_tensor.py

```diff
@@ -695,7 +695,10 @@ def __apply_public_op(
         elif op_str in {"add", "sub"}:
             shares = list(self.share_ptrs)
             # Only the rank 0 party has to add the element
-            shares[0] = op(shares[0], y)
+            if self.session.protocol.share_class == ShareTensor:
+                shares[0] = op(shares[0], y)
+            else:
```

Comment:
`elif self.session.protocol.share_class == ReplicatedSharedTensor` and then else: raise TypeError. Like I mentioned in one of my previous PR's a good practice to throw an error rather than having it silently fail. For example, what if public_op function accidentally gets called somewhere when y is a MPCTensor. This kind of error have occurred when I was working on development.